src/main/java/user_interface/default_game/GamePanel.java

```diff
@@ -23,6 +24,12 @@ public void createGamePanelUI() {
         window.pack();
         window.setLocationRelativeTo(null);
         window.setVisible(true);
+
+        presenter.startGameThread();
+
+        GamePanelController controller = new GamePanelController();
+        window.addKeyListener(controller);
+        window.setFocusable(true);
     }
 
 }
```

Comment:
extra space

Reply:
that startGameThread() method call got me excited I'm not going to lie :D